neuraxle/metaopt/trial.py

```diff
@@ -295,7 +301,15 @@ def get_best_hyperparams(self) -> HyperparameterSamples:
 
         for trial in self.trials:
             trial_score = trial.metrics_results['main']['higher_score_is_better']
-            if best_score is None or higher_score_is_better == (trial_score > best_score):
+            if best_score is None:
+                best_score = trial_score
+                best_hyperparams = trial.hyperparams
+
+            if trial_score > best_score and higher_score_is_better:
+                best_score = trial_score
+                best_hyperparams = trial.hyperparams
+
+            if trial_score < best_score and not higher_score_is_better:
```

Comment:
There might be a way to combine this if with the previous one: `(trial_score > best_score) == higher_score_is_better`. 

Optional fix, you can resolve. 

Note: I'd prefer `<=` rather than `<` but that is of not much importance here.